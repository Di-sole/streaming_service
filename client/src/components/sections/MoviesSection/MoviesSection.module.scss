.section {
    &__header {
        display: inline-block;
        margin-bottom: 16px;
    }

    &__title {
        font-size: 24px;
        font-weight: 600;
        color: #ffffff;
    }
}
