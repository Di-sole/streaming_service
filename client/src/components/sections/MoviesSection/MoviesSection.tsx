import Slider from '@/components/Slider/Slider';
import IMovies from '@/models/IMovies';
import PostersList from '@/components/posters/PostersList/PostersList';
import ArrowedLink from '@/components/UI/links/ArrowedLink/ArrowedLink';
import styles from './MoviesSection.module.scss';
import CardLink from '@/components/UI/links/CardLink/CardLink';

interface MoviesSectionProps {
    title: string;
    movies: IMovies[];
    href: string;
    showAllLink?: boolean;
}

const MoviesSection = ({ title, movies, href, showAllLink = true }: MoviesSectionProps) => {
    return (
        <div className={styles.section}>
            <div className={styles.section__header}>
                {href ? <ArrowedLink text={title} href={href} /> : <h3 className={styles.section__title}>{title}</h3>}
            </div>

            <div className={styles.section__content}>
                <Slider itemType="preview" length={movies.length + 1}>
                    <PostersList posterType="preview" movies={movies} />
                    {showAllLink && <CardLink href={href}>Посмотреть все</CardLink>}
                </Slider>
            </div>
        </div>
    );
};

export default MoviesSection;
