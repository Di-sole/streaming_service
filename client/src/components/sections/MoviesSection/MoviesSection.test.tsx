import { render, screen } from '@testing-library/react';
import MoviesSection from './MoviesSection';
import { moviesTest } from '../../../testAsserts/testItems';

jest.mock('next/router', () => ({
    useRouter() {
        return {
            pathname: '',
        };
    },
}));

describe('MOVIESSECTION SNAPSHOTS TESTS', () => {
    test('MoviesSection should not be changed', () => {
        render(<MoviesSection movies={moviesTest} title="title" href="/" />);

        const id = screen.getByTestId('moviesSection');
        expect(id).toMatchSnapshot();
    });
});
