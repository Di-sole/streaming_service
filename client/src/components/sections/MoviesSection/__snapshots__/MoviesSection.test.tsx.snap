// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MOVIESSECTION SNAPSHOTS TESTS MoviesSection should not be changed 1`] = `
<div
  class="section"
  data-testid="moviesSection"
>
  <div
    class="section__header"
  >
    <a
      class="link"
      href="/"
    >
      <h3>
        title
      </h3>
      <img
        alt=""
        class="arrow"
        data-nimg="1"
        decoding="async"
        height="40"
        loading="lazy"
        src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
        srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
        style="color: transparent;"
        width="40"
      />
    </a>
  </div>
  <div
    class="section__content"
  >
    <div
      class="jsx-687504307 container container_preview"
      data-testid="slider"
    >
      <div
        class="jsx-687504307 container__contentContainer"
      >
        <div
          class="jsx-687504307 container__content content container__content_preview"
        >
          <a
            class="container previewItem"
            href="/movies/535341"
          >
            <div
              class="container__block container__block_image"
            >
              <div
                class="container__imageWrapper"
              >
                <img
                  alt="1+1"
                  class="container__image"
                  data-nimg="fill"
                  decoding="async"
                  loading="lazy"
                  sizes="100vw"
                  src="/_next/image?url=https%3A%2F%2Fst.kp.yandex.net%2Fimages%2Ffilm_iphone%2Fiphone360_535341.jpg&w=3840&q=75"
                  srcset="/_next/image?url=https%3A%2F%2Fst.kp.yandex.net%2Fimages%2Ffilm_iphone%2Fiphone360_535341.jpg&w=640&q=75 640w, /_next/image?url=https%3A%2F%2Fst.kp.yandex.net%2Fimages%2Ffilm_iphone%2Fiphone360_535341.jpg&w=750&q=75 750w, /_next/image?url=https%3A%2F%2Fst.kp.yandex.net%2Fimages%2Ffilm_iphone%2Fiphone360_535341.jpg&w=828&q=75 828w, /_next/image?url=https%3A%2F%2Fst.kp.yandex.net%2Fimages%2Ffilm_iphone%2Fiphone360_535341.jpg&w=1080&q=75 1080w, /_next/image?url=https%3A%2F%2Fst.kp.yandex.net%2Fimages%2Ffilm_iphone%2Fiphone360_535341.jpg&w=1200&q=75 1200w, /_next/image?url=https%3A%2F%2Fst.kp.yandex.net%2Fimages%2Ffilm_iphone%2Fiphone360_535341.jpg&w=1920&q=75 1920w, /_next/image?url=https%3A%2F%2Fst.kp.yandex.net%2Fimages%2Ffilm_iphone%2Fiphone360_535341.jpg&w=2048&q=75 2048w, /_next/image?url=https%3A%2F%2Fst.kp.yandex.net%2Fimages%2Ffilm_iphone%2Fiphone360_535341.jpg&w=3840&q=75 3840w"
                  style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; color: transparent;"
                />
              </div>
              <div
                class="container"
                data-testid="previewPosterContent"
              >
                <div
                  class="container__icons"
                >
                  <img
                    alt="icon"
                    class="container__icon container__icon_favorite"
                    data-nimg="1"
                    decoding="async"
                    height="40"
                    loading="lazy"
                    src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
                    srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
                    style="color: transparent;"
                    title="Смотреть позже"
                    width="40"
                  />
                  <img
                    alt="icon"
                    class="container__icon container__icon_similar"
                    data-nimg="1"
                    decoding="async"
                    height="40"
                    loading="lazy"
                    src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
                    srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
                    style="color: transparent;"
                    title="Похожее"
                    width="40"
                  />
                  <img
                    alt="icon"
                    class="container__icon container__icon_rate"
                    data-nimg="1"
                    decoding="async"
                    height="40"
                    loading="lazy"
                    src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
                    srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
                    style="color: transparent;"
                    title="Уже смотрел, оценить"
                    width="40"
                  />
                  <img
                    alt="icon"
                    class="container__icon container__icon_dislike"
                    data-nimg="1"
                    decoding="async"
                    height="40"
                    loading="lazy"
                    src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
                    srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
                    style="color: transparent;"
                    title="Не нравится такое"
                    width="40"
                  />
                </div>
                <div
                  class="container container__info"
                  data-testid="previewPosterContentBrief"
                >
                  <div
                    class="container__rating"
                  >
                    <span
                      class="container__rating container__rating_integer"
                    >
                      8
                    </span>
                    <span
                      class="container__rating container__rating_fraction"
                    >
                      .8
                    </span>
                  </div>
                  <div
                    class="container__text"
                  >
                    <p>
                      2011
                      , 
                      France
                      ,
                       
                      Drama
                    </p>
                    <p>
                      112 minutes.2
                    </p>
                  </div>
                </div>
              </div>
              <div
                class="badge container__badge"
                data-testid="ageBadge"
              >
                16
                +
              </div>
            </div>
            <div
              class="container__block container__block_text"
            >
              <p
                class="container__title"
              >
                1+1
              </p>
            </div>
          </a>
          <a
            class="container previewItem"
            href="/movies/1143242"
          >
            <div
              class="container__block container__block_image"
            >
              <div
                class="container__imageWrapper"
              >
                <img
                  alt="Джентльмены"
                  class="container__image"
                  data-nimg="fill"
                  decoding="async"
                  loading="lazy"
                  sizes="100vw"
                  src="/_next/image?url=https%3A%2F%2Fst.kp.yandex.net%2Fimages%2Ffilm_iphone%2Fiphone360_1143242.jpg&w=3840&q=75"
                  srcset="/_next/image?url=https%3A%2F%2Fst.kp.yandex.net%2Fimages%2Ffilm_iphone%2Fiphone360_1143242.jpg&w=640&q=75 640w, /_next/image?url=https%3A%2F%2Fst.kp.yandex.net%2Fimages%2Ffilm_iphone%2Fiphone360_1143242.jpg&w=750&q=75 750w, /_next/image?url=https%3A%2F%2Fst.kp.yandex.net%2Fimages%2Ffilm_iphone%2Fiphone360_1143242.jpg&w=828&q=75 828w, /_next/image?url=https%3A%2F%2Fst.kp.yandex.net%2Fimages%2Ffilm_iphone%2Fiphone360_1143242.jpg&w=1080&q=75 1080w, /_next/image?url=https%3A%2F%2Fst.kp.yandex.net%2Fimages%2Ffilm_iphone%2Fiphone360_1143242.jpg&w=1200&q=75 1200w, /_next/image?url=https%3A%2F%2Fst.kp.yandex.net%2Fimages%2Ffilm_iphone%2Fiphone360_1143242.jpg&w=1920&q=75 1920w, /_next/image?url=https%3A%2F%2Fst.kp.yandex.net%2Fimages%2Ffilm_iphone%2Fiphone360_1143242.jpg&w=2048&q=75 2048w, /_next/image?url=https%3A%2F%2Fst.kp.yandex.net%2Fimages%2Ffilm_iphone%2Fiphone360_1143242.jpg&w=3840&q=75 3840w"
                  style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; color: transparent;"
                />
              </div>
              <div
                class="container"
                data-testid="previewPosterContent"
              >
                <div
                  class="container__icons"
                >
                  <img
                    alt="icon"
                    class="container__icon container__icon_favorite"
                    data-nimg="1"
                    decoding="async"
                    height="40"
                    loading="lazy"
                    src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
                    srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
                    style="color: transparent;"
                    title="Смотреть позже"
                    width="40"
                  />
                  <img
                    alt="icon"
                    class="container__icon container__icon_similar"
                    data-nimg="1"
                    decoding="async"
                    height="40"
                    loading="lazy"
                    src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
                    srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
                    style="color: transparent;"
                    title="Похожее"
                    width="40"
                  />
                  <img
                    alt="icon"
                    class="container__icon container__icon_rate"
                    data-nimg="1"
                    decoding="async"
                    height="40"
                    loading="lazy"
                    src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
                    srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
                    style="color: transparent;"
                    title="Уже смотрел, оценить"
                    width="40"
                  />
                  <img
                    alt="icon"
                    class="container__icon container__icon_dislike"
                    data-nimg="1"
                    decoding="async"
                    height="40"
                    loading="lazy"
                    src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
                    srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
                    style="color: transparent;"
                    title="Не нравится такое"
                    width="40"
                  />
                </div>
                <div
                  class="container container__info"
                  data-testid="previewPosterContentBrief"
                >
                  <div
                    class="container__rating"
                  >
                    <span
                      class="container__rating container__rating_integer"
                    >
                      8
                    </span>
                    <span
                      class="container__rating container__rating_fraction"
                    >
                      .5
                    </span>
                  </div>
                  <div
                    class="container__text"
                  >
                    <p>
                      2019
                      , 
                      Great Britain
                      ,
                       
                      Crime
                    </p>
                    <p>
                      113 minutes.2
                    </p>
                  </div>
                </div>
              </div>
              <div
                class="badge container__badge"
                data-testid="ageBadge"
              >
                18
                +
              </div>
            </div>
            <div
              class="container__block container__block_text"
            >
              <p
                class="container__title"
              >
                Джентльмены
              </p>
            </div>
          </a>
          <a
            class="container previewItem"
            href="/movies/462682"
          >
            <div
              class="container__block container__block_image"
            >
              <div
                class="container__imageWrapper"
              >
                <img
                  alt="Волк с Уолл-стрит"
                  class="container__image"
                  data-nimg="fill"
                  decoding="async"
                  loading="lazy"
                  sizes="100vw"
                  src="/_next/image?url=https%3A%2F%2Fst.kp.yandex.net%2Fimages%2Ffilm_iphone%2Fiphone360_462682.jpg&w=3840&q=75"
                  srcset="/_next/image?url=https%3A%2F%2Fst.kp.yandex.net%2Fimages%2Ffilm_iphone%2Fiphone360_462682.jpg&w=640&q=75 640w, /_next/image?url=https%3A%2F%2Fst.kp.yandex.net%2Fimages%2Ffilm_iphone%2Fiphone360_462682.jpg&w=750&q=75 750w, /_next/image?url=https%3A%2F%2Fst.kp.yandex.net%2Fimages%2Ffilm_iphone%2Fiphone360_462682.jpg&w=828&q=75 828w, /_next/image?url=https%3A%2F%2Fst.kp.yandex.net%2Fimages%2Ffilm_iphone%2Fiphone360_462682.jpg&w=1080&q=75 1080w, /_next/image?url=https%3A%2F%2Fst.kp.yandex.net%2Fimages%2Ffilm_iphone%2Fiphone360_462682.jpg&w=1200&q=75 1200w, /_next/image?url=https%3A%2F%2Fst.kp.yandex.net%2Fimages%2Ffilm_iphone%2Fiphone360_462682.jpg&w=1920&q=75 1920w, /_next/image?url=https%3A%2F%2Fst.kp.yandex.net%2Fimages%2Ffilm_iphone%2Fiphone360_462682.jpg&w=2048&q=75 2048w, /_next/image?url=https%3A%2F%2Fst.kp.yandex.net%2Fimages%2Ffilm_iphone%2Fiphone360_462682.jpg&w=3840&q=75 3840w"
                  style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; color: transparent;"
                />
              </div>
              <div
                class="container"
                data-testid="previewPosterContent"
              >
                <div
                  class="container__icons"
                >
                  <img
                    alt="icon"
                    class="container__icon container__icon_favorite"
                    data-nimg="1"
                    decoding="async"
                    height="40"
                    loading="lazy"
                    src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
                    srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
                    style="color: transparent;"
                    title="Смотреть позже"
                    width="40"
                  />
                  <img
                    alt="icon"
                    class="container__icon container__icon_similar"
                    data-nimg="1"
                    decoding="async"
                    height="40"
                    loading="lazy"
                    src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
                    srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
                    style="color: transparent;"
                    title="Похожее"
                    width="40"
                  />
                  <img
                    alt="icon"
                    class="container__icon container__icon_rate"
                    data-nimg="1"
                    decoding="async"
                    height="40"
                    loading="lazy"
                    src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
                    srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
                    style="color: transparent;"
                    title="Уже смотрел, оценить"
                    width="40"
                  />
                  <img
                    alt="icon"
                    class="container__icon container__icon_dislike"
                    data-nimg="1"
                    decoding="async"
                    height="40"
                    loading="lazy"
                    src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
                    srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
                    style="color: transparent;"
                    title="Не нравится такое"
                    width="40"
                  />
                </div>
                <div
                  class="container container__info"
                  data-testid="previewPosterContentBrief"
                >
                  <div
                    class="container__rating"
                  >
                    <span
                      class="container__rating container__rating_integer"
                    >
                      8
                    </span>
                    <span
                      class="container__rating container__rating_fraction"
                    >
                      .0
                    </span>
                  </div>
                  <div
                    class="container__text"
                  >
                    <p>
                      2013
                      , 
                      USA
                      ,
                       
                      Drama
                    </p>
                    <p>
                      180 minutes.2
                    </p>
                  </div>
                </div>
              </div>
              <div
                class="badge container__badge"
                data-testid="ageBadge"
              >
                18
                +
              </div>
            </div>
            <div
              class="container__block container__block_text"
            >
              <p
                class="container__title"
              >
                Волк с Уолл-стрит
              </p>
            </div>
          </a>
          <a
            class="container previewItem"
            href="/movies/251733"
          >
            <div
              class="container__block container__block_image"
            >
              <div
                class="container__imageWrapper"
              >
                <img
                  alt="Аватар"
                  class="container__image"
                  data-nimg="fill"
                  decoding="async"
                  loading="lazy"
                  sizes="100vw"
                  src="/_next/image?url=https%3A%2F%2Fst.kp.yandex.net%2Fimages%2Ffilm_iphone%2Fiphone360_251733.jpg&w=3840&q=75"
                  srcset="/_next/image?url=https%3A%2F%2Fst.kp.yandex.net%2Fimages%2Ffilm_iphone%2Fiphone360_251733.jpg&w=640&q=75 640w, /_next/image?url=https%3A%2F%2Fst.kp.yandex.net%2Fimages%2Ffilm_iphone%2Fiphone360_251733.jpg&w=750&q=75 750w, /_next/image?url=https%3A%2F%2Fst.kp.yandex.net%2Fimages%2Ffilm_iphone%2Fiphone360_251733.jpg&w=828&q=75 828w, /_next/image?url=https%3A%2F%2Fst.kp.yandex.net%2Fimages%2Ffilm_iphone%2Fiphone360_251733.jpg&w=1080&q=75 1080w, /_next/image?url=https%3A%2F%2Fst.kp.yandex.net%2Fimages%2Ffilm_iphone%2Fiphone360_251733.jpg&w=1200&q=75 1200w, /_next/image?url=https%3A%2F%2Fst.kp.yandex.net%2Fimages%2Ffilm_iphone%2Fiphone360_251733.jpg&w=1920&q=75 1920w, /_next/image?url=https%3A%2F%2Fst.kp.yandex.net%2Fimages%2Ffilm_iphone%2Fiphone360_251733.jpg&w=2048&q=75 2048w, /_next/image?url=https%3A%2F%2Fst.kp.yandex.net%2Fimages%2Ffilm_iphone%2Fiphone360_251733.jpg&w=3840&q=75 3840w"
                  style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; color: transparent;"
                />
              </div>
              <div
                class="container"
                data-testid="previewPosterContent"
              >
                <div
                  class="container__icons"
                >
                  <img
                    alt="icon"
                    class="container__icon container__icon_favorite"
                    data-nimg="1"
                    decoding="async"
                    height="40"
                    loading="lazy"
                    src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
                    srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
                    style="color: transparent;"
                    title="Смотреть позже"
                    width="40"
                  />
                  <img
                    alt="icon"
                    class="container__icon container__icon_similar"
                    data-nimg="1"
                    decoding="async"
                    height="40"
                    loading="lazy"
                    src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
                    srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
                    style="color: transparent;"
                    title="Похожее"
                    width="40"
                  />
                  <img
                    alt="icon"
                    class="container__icon container__icon_rate"
                    data-nimg="1"
                    decoding="async"
                    height="40"
                    loading="lazy"
                    src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
                    srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
                    style="color: transparent;"
                    title="Уже смотрел, оценить"
                    width="40"
                  />
                  <img
                    alt="icon"
                    class="container__icon container__icon_dislike"
                    data-nimg="1"
                    decoding="async"
                    height="40"
                    loading="lazy"
                    src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
                    srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
                    style="color: transparent;"
                    title="Не нравится такое"
                    width="40"
                  />
                </div>
                <div
                  class="container container__info"
                  data-testid="previewPosterContentBrief"
                >
                  <div
                    class="container__rating"
                  >
                    <span
                      class="container__rating container__rating_integer"
                    >
                      8
                    </span>
                    <span
                      class="container__rating container__rating_fraction"
                    >
                      .0
                    </span>
                  </div>
                  <div
                    class="container__text"
                  >
                    <p>
                      2009
                      , 
                      USA
                      ,
                       
                      Fantasy
                    </p>
                    <p>
                      162 minutes.1
                    </p>
                  </div>
                </div>
              </div>
              <div
                class="badge container__badge"
                data-testid="ageBadge"
              >
                12
                +
              </div>
            </div>
            <div
              class="container__block container__block_text"
            >
              <p
                class="container__title"
              >
                Аватар
              </p>
            </div>
          </a>
          <a
            class="container previewItem"
            href="/movies/41520"
          >
            <div
              class="container__block container__block_image"
            >
              <div
                class="container__imageWrapper"
              >
                <img
                  alt="Брат 2"
                  class="container__image"
                  data-nimg="fill"
                  decoding="async"
                  loading="lazy"
                  sizes="100vw"
                  src="/_next/image?url=https%3A%2F%2Fst.kp.yandex.net%2Fimages%2Ffilm_iphone%2Fiphone360_41520.jpg&w=3840&q=75"
                  srcset="/_next/image?url=https%3A%2F%2Fst.kp.yandex.net%2Fimages%2Ffilm_iphone%2Fiphone360_41520.jpg&w=640&q=75 640w, /_next/image?url=https%3A%2F%2Fst.kp.yandex.net%2Fimages%2Ffilm_iphone%2Fiphone360_41520.jpg&w=750&q=75 750w, /_next/image?url=https%3A%2F%2Fst.kp.yandex.net%2Fimages%2Ffilm_iphone%2Fiphone360_41520.jpg&w=828&q=75 828w, /_next/image?url=https%3A%2F%2Fst.kp.yandex.net%2Fimages%2Ffilm_iphone%2Fiphone360_41520.jpg&w=1080&q=75 1080w, /_next/image?url=https%3A%2F%2Fst.kp.yandex.net%2Fimages%2Ffilm_iphone%2Fiphone360_41520.jpg&w=1200&q=75 1200w, /_next/image?url=https%3A%2F%2Fst.kp.yandex.net%2Fimages%2Ffilm_iphone%2Fiphone360_41520.jpg&w=1920&q=75 1920w, /_next/image?url=https%3A%2F%2Fst.kp.yandex.net%2Fimages%2Ffilm_iphone%2Fiphone360_41520.jpg&w=2048&q=75 2048w, /_next/image?url=https%3A%2F%2Fst.kp.yandex.net%2Fimages%2Ffilm_iphone%2Fiphone360_41520.jpg&w=3840&q=75 3840w"
                  style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; color: transparent;"
                />
              </div>
              <div
                class="container"
                data-testid="previewPosterContent"
              >
                <div
                  class="container__icons"
                >
                  <img
                    alt="icon"
                    class="container__icon container__icon_favorite"
                    data-nimg="1"
                    decoding="async"
                    height="40"
                    loading="lazy"
                    src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
                    srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
                    style="color: transparent;"
                    title="Смотреть позже"
                    width="40"
                  />
                  <img
                    alt="icon"
                    class="container__icon container__icon_similar"
                    data-nimg="1"
                    decoding="async"
                    height="40"
                    loading="lazy"
                    src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
                    srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
                    style="color: transparent;"
                    title="Похожее"
                    width="40"
                  />
                  <img
                    alt="icon"
                    class="container__icon container__icon_rate"
                    data-nimg="1"
                    decoding="async"
                    height="40"
                    loading="lazy"
                    src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
                    srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
                    style="color: transparent;"
                    title="Уже смотрел, оценить"
                    width="40"
                  />
                  <img
                    alt="icon"
                    class="container__icon container__icon_dislike"
                    data-nimg="1"
                    decoding="async"
                    height="40"
                    loading="lazy"
                    src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
                    srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
                    style="color: transparent;"
                    title="Не нравится такое"
                    width="40"
                  />
                </div>
                <div
                  class="container container__info"
                  data-testid="previewPosterContentBrief"
                >
                  <div
                    class="container__rating"
                  >
                    <span
                      class="container__rating container__rating_integer"
                    >
                      8
                    </span>
                    <span
                      class="container__rating container__rating_fraction"
                    >
                      .2
                    </span>
                  </div>
                  <div
                    class="container__text"
                  >
                    <p>
                      2000
                      , 
                      Russia
                      ,
                       
                      Action
                    </p>
                    <p>
                      127 minutes.2
                    </p>
                  </div>
                </div>
              </div>
              <div
                class="badge container__badge"
                data-testid="ageBadge"
              >
                16
                +
              </div>
            </div>
            <div
              class="container__block container__block_text"
            >
              <p
                class="container__title"
              >
                Брат 2
              </p>
            </div>
          </a>
          <div>
            <a
              class="container"
              href="/"
            >
              <div
                class="container__content"
              >
                View all
              </div>
            </a>
          </div>
        </div>
      </div>
      <button
        class="button container__button container__button_right"
      >
        <img
          alt=""
          class="icon icon"
          data-nimg="1"
          decoding="async"
          height="40"
          loading="lazy"
          src="/_next/image?url=%2Fimg.jpg&w=48&q=75"
          srcset="/_next/image?url=%2Fimg.jpg&w=32&q=75 1x, /_next/image?url=%2Fimg.jpg&w=48&q=75 2x"
          style="color: transparent;"
          width="20"
        />
      </button>
    </div>
  </div>
</div>
`;
